pygenomeworks/genomeworks/cuda/cuda.pyx

```diff
@@ -32,40 +33,19 @@ cdef class CudaStream:
     """Class to abstract the usage of CUDA streams which enable
     easy asynchronous execution of CUDA kernels.
     """
-    # Using size_t to store stream since underlying
-    # representation of cudaStream_t is a (void *)
-    # and python doesn't know how to deal with converting
-    # (void *) to python objects.
-    cdef size_t stream
 
-    def __cinit__(self):
-        """Constructs a CudaStream object to encapsulate a CUDA stream.
-        """
-        cdef cuda_runtime._Stream s
-        cdef cuda_runtime._Error e = cuda_runtime.cudaStreamCreate(&s)
-        if (e != 0):
-            raise CudaRuntimeError(e)
-        self.stream = <size_t>s
+    cdef cuda_stream.CudaStream stream
 
     def __init__(self):
-        """Dummy implementation of __init__ function to allow
-        for Python subclassing.
+        """Constructs a CudaStream object to encapsulate a CUDA stream.
         """
+        self.stream = cuda_stream.make_cuda_stream()
```

Comment:
this needs to be in the `__cinit__` function. That's when all C objects need to be initialized

an empty `__init__ object also needs to be kept around to allow for python subclassing

Reply:
I removed all the python changes from this PR.